### PR TITLE
GHA: package up the content as a nuget package

### PR DESCRIPTION
**Pull request by compnerd**
This introduces an initial nuget packaging for simpler distribution of the artifacts.

### CODE REVIEW

File: .github/workflows/bcny-firebase.yml [hunk 1]
```diff
@@ -98,3 +98,142 @@ jobs:
...
+            <metadata>
+              <id>com.google.firebase.windows.${{ matrix.arch }}</id>
+              <version>0.0.0.0</version>
+              <title>Firebase C++ SDK</title>
+              <description>C++ Firebase SDK</description>
+              <authors>Google, Inc.</authors>
+              <projectUrl>https://firebase.google.com</projectUrl>
+              <repository type="git" url="https://github.com/google/firebase-cpp-sdk" branch="main" />
+            </metadata>
+            <files>
```
Comment (brianmichel) [inline .github/workflows/bcny-firebase.yml:116]: Is it possible to put pdbs in here as well? I think these should be getting generated but don't seem to be copied anywhere.

**Review by brianmichel — APPROVED** [review]
LGTM

File: .github/workflows/bcny-firebase.yml (hunk 1 above)
Comment (compnerd) [reply .github/workflows/bcny-firebase.yml:116]: Yes, that would be possible, but I do believe we do not generate them.  We generate the debug information inline in the object file (which is then packaged as part of the static library).

Comment (brianmichel) [reply .github/workflows/bcny-firebase.yml:116]: Sounds good, I have a task to fix symbol info for `swift-firebase` so maybe you can I can pair on that after we ship initially.

Comment (compnerd) [reply .github/workflows/bcny-firebase.yml:116]: Sounds like a plan!

**Pull request merged**